src/compiler/llvm/mir.rs

```diff
@@ -593,6 +620,45 @@ impl<'p, 'module, 'ctx> FunctionBuilder<PointerValue<'ctx>, BasicValueEnum<'ctx>
             .build_conditional_branch(cond.into_int_value(), *then_bb, *else_bb);
     }
 
+    fn term_call_fn(
+        &mut self,
+        target: Location<'ctx>,
+        args: &[BasicValueEnum<'ctx>],
+        reentry: (Location<'ctx>, BasicBlockId),
+    ) {
+        let f = target.into_function().unwrap();
+        let args = match f.ret_method {
+            ReturnMethod::OutParam => {
+                // If the return method is to use an out parameter, then push the
+                // return value location to the front of the argument list for the functoin
+                let out = reentry.0.into_pointer().unwrap().as_basic_value_enum();
+                let mut out_args = vec![out];
+                for a in args {
+                    out_args.push(*a);
+                }
+                out_args
```

Comment:
Not a huge fan of this, is there someway I can prepend the output argument without having to clone the whole argument set?

Perhaps I should make the output argument the last one? Is there any reason not to? One cost would be breaking variadic functions.

Reply:
Add a note in the output parameter about why it is prepended (variadic functions) rather than appended).